schema-loader/src/main/java/com/scalar/db/schemaloader/ImportSchemaParser.java

```diff
@@ -0,0 +1,44 @@
+package com.scalar.db.schemaloader;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class ImportSchemaParser {
+  private final JsonObject schemaJson;
+
+  public ImportSchemaParser(Path jsonFilePath) throws SchemaLoaderException {
+    try (Reader reader = Files.newBufferedReader(jsonFilePath)) {
+      schemaJson = JsonParser.parseReader(reader).getAsJsonObject();
+    } catch (IOException | JsonParseException e) {
+      throw new SchemaLoaderException("Parsing the schema JSON failed", e);
+    }
+  }
+
+  public ImportSchemaParser(String serializedSchemaJson) throws SchemaLoaderException {
+    try {
+      schemaJson = JsonParser.parseString(serializedSchemaJson).getAsJsonObject();
+    } catch (JsonParseException e) {
+      throw new SchemaLoaderException("Parsing the schema JSON failed", e);
+    }
+  }
+
+  public List<ImportTableSchema> parse() throws SchemaLoaderException {
+    ArrayList<ImportTableSchema> tableSchemaList = new ArrayList<>();
```

Comment:
[trivial] We don't need to use the concrete class type `ArrayList` as only `List#add` is used here.

```
    List<ImportTableSchema> tableSchemaList = new ArrayList<>();
```

Reply:
Fixed in 0d3a2f8.